mesonbuild/mcompile.py

```diff
@@ -170,7 +171,7 @@ def get_parsed_args_ninja(options: 'argparse.Namespace', builddir: Path) -> T.Tu
 def generate_target_name_vs(target: ParsedTargetName, builddir: Path, introspect_data: dict) -> str:
     intro_target = get_target_from_intro_data(target, builddir, introspect_data)
 
-    assert intro_target['type'] != 'run', 'Should not reach here: `run` targets must be handle above'
+    assert intro_target['type'] not in ('alias', 'run'), 'Should not reach here: `run` targets must be handle above'
```

Comment:
Nitpicking:
```suggestion
    assert intro_target['type'] not in {'alias', 'run'}, 'Should not reach here: `run` targets must be handle above'
```

Other than that, LGTM.

Reply:
oh. wasn't caught by pylint...

Reply:
It would catch if you used a list, which is a lot worse for membership.